### ISSUE TITLE
[DOC] `groupby_agg` naming and documentation

### ISSUE THREAD

**Issue by jiafengkevinchen** [ericmjl/pyjanitor#477]
`groupby_agg` is possible to implement with method-chaining in native pandas, contrary to what the documentation seems to imply; native pandas would use `groupby(...).transform`, e.g.
```python
df = df.assign(new_column=lambda x: x.groupby("col")["col_to_agg"].transform(agg_func))
```
Of course, this is a mouthful to type, so it still makes sense to have a function that takes care of this operation, but `groupby_agg` seems like the wrong name for the function, and something like `groupby_transform_new_column` would be more descriptive. 

I'm not sure whether the implementation should also change to reflect that native pandas supports `groupby(...).transform`, and so the `.merge` would not be necessary.

**Comment by ericmjl** [comment]
@jiafengkevinchen thanks for raising this! Your points are well-taken. Would you like to work on a PR to improve the docs? Specifically, the scope of the PR would be to replace the current “pandas example” with what you provided, alongside a sentence that describes that one might find it tedious to type.

On the implementation, I’m not too concerned, unless there are clear performance (memory/time/maintenance) benefits. If you dig in and find that using your method improves on any two of the three criteria, I’d be happy to accept that as a PR as well.

And finally, on the function name, the hard part about APIs is coming up with a good name; changing it can break others’ code, so I’d be hesitant to change the name. 

In general, we try to strive for function names that are short (two-word) and descriptive, while also being able to be read left-to-right while sounding English-like. (As an example, `.transform_column(column_name=“col1”, func=func, new_column_name=“col2”)` reads as ‘transform column “col1” through this function into a new column “col2”’.)

In the case of the `groupby_agg` case, it’s hard to come up with a two-word verb-like phrase that describes what is going on, so I think the original creator of the function simply went by the primary/most important operations going on under the hood. I recognized that, and let the descriptive criteria go in this case. 

Regardless, the API docs can be thought of as an English dictionary - and we do have weird words in English for which their etymology can be weird, so I’m not too worried about this name. Hope that provides some context.

**Comment by jiafengkevinchen** [comment]
There is a performance benefit actually (due to @frtennis1)
![image](https://user-images.githubusercontent.com/24930289/61420402-407b5800-a8d0-11e9-8bba-fc7c34bcb0d0.png)

**Comment by ericmjl** [comment]
Superb! Looking forward to reviewing your PR after work today.

**Comment by ericmjl** [comment]
Closed by #480.
